frontend/views/pages/Contributions.vue

```diff
@@ -129,25 +197,66 @@ export default {
       }
     }
   },
+  validations: {
+    form: {
+      incomeAmount: { required, minValue: v => v >= 0, decimals: decimals(2) },
+      pledgeAmount: { required, minValue: v => v >= 0, decimals: decimals(2) }
+    }
+  },
   computed: {
+    ...mapGetters([
+      'groupSettings',
+      'memberProfile'
+    ]),
+    currency () {
+      return currencies[this.groupSettings.mincomeCurrency]
+    },
     doesReceiveMonetary () {
       return !!this.fakeStore.receiving.monetary && !this.ephemeral.isEditingIncome
     },
     doesGiveMonetary () {
       return !!this.fakeStore.giving.monetary && !this.ephemeral.isEditingIncome
     }
   },
+  mounted () {
+    const profile = this.memberProfile(this.$store.state.loggedIn.username)
+    const incomeDetailsKey = profile && profile.groupProfile && profile.groupProfile.incomeDetailsKey
```

Comment:
could we do this before mounted cycle? I believe it should be on created or beforeMount. that way we don't re-render the component twice.